Modules/_io/bufferedio.c

```diff
@@ -1301,6 +1287,22 @@ _io__Buffered_seek_impl(buffered *self, PyObject *targetobj, int whence)
     return res;
 }
 
+static PyObject *
+buffered_tell(buffered *self, PyObject *Py_UNUSED(ignored))
+{
+    /* buffered_tell() is implemented as a wrappper
+       around _io__Buffered_seek_impl() because it
+       keeps the location information properly
+       updated. The optimizations that had previously
+       been implemented are prone to get out of sync
+       with the kernel when the file has been opened
+       in append mode. See issue #36411.
+    */
+
+    CHECK_INITIALIZED(self)
+    return _io__Buffered_seek_impl(self, PyLong_FromLong(0), SEEK_CUR);
```

Comment:
For what it's worth, I believe `PyLong_FromLong(0)` should be checked for failure (even though this call should never fail due to the current implementation of `PyLong_FromLong()`).

Reply:
> For what it's worth, I believe `PyLong_FromLong(0)` should be checked for failure (even though this call should never fail due to the current implementation of `PyLong_FromLong()`).

I agree, and made the change.